### DISCUSSION TITLE
Missing PATH entries

### DISCUSSION THREAD

**Discussion by fmorroni** [awesomeWM/awesome#3936] [Q&A]
Hi, I just noticed that the `~/.local/bin` entry is missing from the PATH within awesome. All the entries I've manually added from .zshenv appear without issue, but `~/.local/bin` which is automatically added by the system doesn't appear. I'm not sure where exactly this entry gets included into the PATH, but I'm pretty sure it should be before awesome gets loaded, so idk why it's not being detected. Any ideas?

Of course I could just add `~/.local/bin` to .zshenv and that should work, but I don't want to have repeated entries and I want to identify the problem anyways.

I'm on arch linux in case that matters. Thanks!

**Comment by fmorroni** [comment]
Hi @actionless sorry to bother you, but I can't seem to find which discussion/issue my question is duplicating. I did search before posting but the issues I found where relating to env vars added to .zshrc/.bashrc

**Reply by actionless** [reply]
and what is PATH?

**Reply by fmorroni** [reply]
I mean the environment variable containing the paths from where binaries are searched. I have some executables in `~/.local/bin` which are not found when running scripts from within awesome.

**Reply by actionless** [reply]
>  I did search before posting but the issues I found where relating to env vars

> I mean the environment variable

🤦

**Reply by fmorroni** [reply]
What's the point of cutting a quote short just to change the apparent meaning? As I said, the original issue talked about missing entries that where added to the path through .bashrc/.zshrc, which makes sense because those files get sourced after awesome starts. 
My problem is that an entry that gets added **before** awesome starts (~/.local/bin) is missing from the path env var of executables run through awesome.

**Reply by actionless** [reply]
no, that's still the same thing

**Comment by sclu1034** [comment]
The `bash` manual explains why and when certain files are sourced and when they aren't: https://man.archlinux.org/man/core/bash/bash.1.en#INVOCATION. The same concept applies to `zsh` and other shells.

**Reply by fmorroni** [reply]
Hi thanks I'll give it a look tomorrow.
